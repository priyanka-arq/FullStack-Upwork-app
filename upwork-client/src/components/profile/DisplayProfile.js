import React, { useEffect } from "react";
import { CloudinaryContext, Transformation, Image } from "cloudinary-react";
import phoneImage from "../Images/phoneImage.png";
import emailImage from "../Images/emailImage1.png";
import gitImage from "../Images/gitImage.png";
import AuthService from "../../services/auth.service";

const DisplayProfile = (props) => {
  const [disable, setDisable] = React.useState(true);
  const [currentUser, setCurrentUser] = React.useState({});

  useEffect(() => {
    const user = AuthService.getCurrentUser();
    setCurrentUser(user);
    if (user && user.id === props.profile.user) {
      setDisable(false);
    } else {
      setDisable(true);
    }
  }, [props.profile]);

  return (
    <div className="container">
      <div className="profileHeading">
        <h1>{props.profile.name}</h1>
        <h3>{props.profile.title}</h3>
      </div>

      <hr />
      <div className="aboutMe">
        <CloudinaryContext cloudName="didtkbpn7">
          <Image className="headshot" publicId={props.profile.imageUrl}>
            <Transformation
              crop="scale"
              width="300"
              height="200"
              dpr="auto"
              responsive_placeholder="blank"
            />
          </Image>
        </CloudinaryContext>
        <div>
          <img
            id="phoneImage"
            src={phoneImage}
            alt={props.profile.phone}
            height="20"
            width="20"
          />
          <p id="phone" className="contact">
            {props.profile.phone}
          </p>
        </div>

        <div>
          <img
            id="emailImage"
            src={emailImage}
            alt={props.profile.email}
            height="20"
            width="20"
          />
          <p id="email" className="contact" href={props.profile.email}>
            {props.profile.email}
          </p>
        </div>

        <div>
          <img
            id="gitImage"
            src={gitImage}
            alt={props.profile.github}
            height="20"
            width="20"
          />
          <a
            id="git"
            className="contact"
            target="_blank"
            href={props.profile.github}
          >
            {props.profile.github}
          </a>
        </div>
      </div>

      <div className="profileInfo">
        <h1>Hello,</h1>
        <h2>A bit more about me: </h2>
        <h5>{props.profile.aboutme}</h5>
        <div>
          {props.profile.skills && (
            <div className="skills">
              <h4> Professional Skills:</h4>
              {props.profile.skills.map((skill) => (
                <span>{skill}</span>
              ))}
            </div>
          )}
          <div className="resume">
            <p>
              <a href={props.profile.resume} target="_blank"></a>Resume
            </p>
          </div>

          <div className="linkedIn">
            <p>
              <a href={props.profile.linkedIn} target="_blank">
                LinkedIn
              </a>
            </p>
          </div>

          <div className="portfolio">
            <p>
              <a href={props.profile.portfolio} target="_blank">
                Portfolio
              </a>
            </p>
          </div>
        </div>
      </div>

      <div>
        <button
          className="button"
          onClick={props.handleUpdate}
          disabled={disable}
        >
          Update Profile
        </button>
      </div>
    </div>
  );
};

export default DisplayProfile;
