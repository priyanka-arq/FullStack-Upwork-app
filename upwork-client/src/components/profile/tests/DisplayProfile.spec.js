import React from "react";
import DisplayProfile from "../DisplayProfile";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import profiles from "../../../fixtures/profiles";

let wrapped;
beforeEach(() => {
  wrapped = shallow(<DisplayProfile profile={profiles[1]} />);
});

afterEach(() => {
  wrapped.unmount();
});

test("should render Display Profile correctly", () => {
  expect(toJSON(wrapped)).toMatchSnapshot();
});

it("has a button and img", () => {
  //test jsx elements
  expect(wrapped.find("button").length).toEqual(1);
  expect(wrapped.find("img").length).toEqual(3);

  //css selector
  expect(wrapped.find(".portfolio").length).toEqual(1);

  //expect(wrapped.props().profile).toEqual("profile");
});
