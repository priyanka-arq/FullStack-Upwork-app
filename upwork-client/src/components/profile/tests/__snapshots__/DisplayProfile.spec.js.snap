// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Display Profile correctly 1`] = `
<div
  className="container"
>
  <div
    className="profileHeading"
  >
    <h1>
      Test1
    </h1>
    <h3>
      Developer
    </h3>
  </div>
  <hr />
  <div
    className="aboutMe"
  >
    <CloudinaryContext
      cloudName="didtkbpn7"
      includeOwnBody={false}
    >
      <Image
        className="headshot"
        publicId={null}
      >
        <CloudinaryTransformation
          crop="scale"
          dpr="auto"
          height="200"
          responsive_placeholder="blank"
          width="300"
        />
      </Image>
    </CloudinaryContext>
    <div>
      <img
        alt="04444444444"
        height="20"
        id="phoneImage"
        src="phoneImage.png"
        width="20"
      />
      <p
        className="contact"
        id="phone"
      >
        04444444444
      </p>
    </div>
    <div>
      <img
        alt="test1@ga.com"
        height="20"
        id="emailImage"
        src="emailImage1.png"
        width="20"
      />
      <p
        className="contact"
        href="test1@ga.com"
        id="email"
      >
        test1@ga.com
      </p>
    </div>
    <div>
      <img
        alt="https://github.com/little-chi-mai"
        height="20"
        id="gitImage"
        src="gitImage.png"
        width="20"
      />
      <a
        className="contact"
        href="https://github.com/little-chi-mai"
        id="git"
        target="_blank"
      >
        https://github.com/little-chi-mai
      </a>
    </div>
  </div>
  <div
    className="profileInfo"
  >
    <h1>
      Hello,
    </h1>
    <h2>
      A bit more about me: 
    </h2>
    <h5>
      I have a great passion about solving real world problems and helping people. And now, with the skills that I have been learning in programming and computer science from the Software Engineering Course at General Assembly, I will be able to do that with new technologies. I believe in the power of technology and its impact to everyone's daily lives.
    </h5>
    <div>
      <div
        className="skills"
      >
        <h4>
           Professional Skills:
        </h4>
        <span>
          HTML
        </span>
        <span>
          PHP
        </span>
        <span>
          Java
        </span>
        <span>
          CSS
        </span>
      </div>
      <div
        className="resume"
      >
        <p>
          <a
            href=""
            target="_blank"
          />
          Resume
        </p>
      </div>
      <div
        className="linkedIn"
      >
        <p>
          <a
            href="https://linkedin.com/in/chi-mai-nguyenn"
            target="_blank"
          >
            LinkedIn
          </a>
        </p>
      </div>
      <div
        className="portfolio"
      >
        <p>
          <a
            href="https://little-chi-mai.github.io/portfolio"
            target="_blank"
          >
            Portfolio
          </a>
        </p>
      </div>
    </div>
  </div>
  <div>
    <button
      className="button"
      disabled={true}
    >
      Update Profile
    </button>
  </div>
</div>
`;
